src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1323,6 +1330,71 @@ private void SetTotalDifficulty(BlockHeader header)
 
         public event EventHandler<BlockEventArgs> NewHeadBlock;
 
+        public void DeleteChainSlice(in long startNumber, in long endNumber)
+        {
+            if (endNumber - startNumber < 0)
+            {
+                throw new InvalidOperationException($"Start number must be equal or greater end number.");
+            }
+            
+            if (endNumber - startNumber > 1000)
+            {
+                throw new InvalidOperationException($"Cannot delete that many blocks at once (start: {startNumber}, end {endNumber}).");
+            }
+            
+            if (endNumber < BestKnownNumber)
```

Comment:
We can only delete what? Levels we above the ones we know? I don't get this condition.

Reply:
a slice has to start before the end and end after the end (in case there are corrupted levels in the future)

Reply:
added a comment to the method

Reply:
Why can't we specify arbitrary large endNumber (longer than BestKnownNumber) and then delete only up to BestKnownNumber? Maybe it doesn't make sense though. Also shouldn't it be <= rather than < ? So we can delete up to BestKnownNumber inclusive?

Reply:
because this is to handle corrupted state where BestKnownNumber may be wrong

Reply:
still shouldn't it be <= ?